internal/services/managedredis/managed_redis_access_policy_assignment_resource.go

```diff
@@ -0,0 +1,187 @@
+
+			existing, err := client.AccessPolicyAssignmentGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			dbResp, err := client.Get(ctx, dbId)
+			if err != nil {
+				if response.WasNotFound(dbResp.HttpResponse) {
+					return fmt.Errorf("managed Redis database %s was not found", dbId)
+				}
+				return fmt.Errorf("retrieving %s: %+v", dbId, err)
+			}
+
+			createInput := databases.AccessPolicyAssignment{
+				Name: pointer.To(model.ObjectID),
+				Properties: &databases.AccessPolicyAssignmentProperties{
+					AccessPolicyName: "default",
+					User: databases.AccessPolicyAssignmentPropertiesUser{
+						ObjectId: pointer.To(model.ObjectID),
+					},
+				},
+			}
+
+			locks.ByID(model.ManagedRedisID)
+			defer locks.UnlockByID(model.ManagedRedisID)
+
+			if err := client.AccessPolicyAssignmentCreateUpdateThenPoll(ctx, id, createInput); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := databases.ParseAccessPolicyAssignmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.ManagedRedis.DatabaseClient
+
+			resp, err := client.AccessPolicyAssignmentGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := ManagedRedisAccessPolicyAssignmentResourceModel{}
+
+			if model := resp.Model; model != nil {
+				clusterId := redisenterprise.NewRedisEnterpriseID(id.SubscriptionId, id.ResourceGroupName, id.RedisEnterpriseName)
+				state.ManagedRedisID = clusterId.ID()
+
+				if props := model.Properties; props != nil {
+					if props.User.ObjectId != nil {
+						state.ObjectID = *props.User.ObjectId
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r ManagedRedisAccessPolicyAssignmentResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedRedisAccessPolicyAssignmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.ManagedRedis.DatabaseClient
+			id, err := databases.ParseAccessPolicyAssignmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			locks.ByID(model.ManagedRedisID)
+			defer locks.UnlockByID(model.ManagedRedisID)
```

Comment:
We usually don't need this if the API has rate control. It's only necessary when the API can't create two resources simultaneously. If testing shows the API can create two Managed Redis Access Policy resources at once, we can remove it. Could you confirm?